sdk/mobile-center-distribute/src/main/java/com/microsoft/azure/mobile/distribute/Distribute.java

```diff
@@ -1119,6 +1127,39 @@ private void viewReleaseNotes(ReleaseDetails releaseDetails) {
     }
 
     /**
+     * Store failed package info hash in preferences.
+     */
+    private synchronized void storeUpdateSetupFailedPackageHash() {
+        if (mPackageInfo != null && mUpdateSetupFailedDialog != null) {
```

Comment:
If mPackageInfo is null, that means the SDK is not started, that does not mean the SDK is disabled.
Please look at logic around `mBeforeStartUpdateToken` that addresses this scenario.